### PR TITLE
build: unbundle pester

### LINKED ISSUE #5458: Unbundle Pester from PSCore pkg
We originally decided to ship Pester inbox in Windows to help bootstrap the ecosystem with a common test framework.  I believe we’ve achieved that goal.  We should stop shipping Pester with PSCore6 and rely on users installing it from PSGallery.  Long term we should move other modules we ship/own to be available on PSGallery and compatible with a range of PSCore6 versions.  This should happen after we update our tests to work on latest Pester

### PR DESCRIPTION
**Pull request by TravisEz13**
fixes #5458
* remove pester submodule
* add a `-CI` switch to `start-PSBuild`
* only add pester if `-CI` is specified and `-PSModuleRestore`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ script:
   - pwsh -File tools/travis.ps1
   # spellcheck
   - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
-      mdspell '**/*.md' '!powershell/**/*.md' --ignore-numbers --ignore-acronyms --report;
+      mdspell '**/*.md' '!powershell/**/*.md' '!**/publish/**/*.md' --ignore-numbers --ignore-acronyms --report;
```
Comment (daxian-dbw) [inline .travis.yml]: Can `'!powershell/**/*.md'` maybe be removed?

File: build.psm1 [hunk 2]
```diff
@@ -2320,6 +2327,55 @@ function Clear-PSRepo
...
+        $clonePath = Join-Path -Path $Destination -ChildPath $Name
+        if(Test-Path $clonePath)
+        {
+            remove-Item -Path $clonePath -recurse -force
+        }
+
+        $null = Start-NativeExecution {git clone --quiet $uri $clonePath}
+        Push-location
+        try {
+            Set-Location $clonePath
```
Comment (daxian-dbw) [inline build.psm1]: Minor: maybe use `push-location $clonePath` to combine `push-location` and `set-location $clonePath`

File: build.psm1 [hunk 3]
```diff
@@ -2320,6 +2327,55 @@ function Clear-PSRepo
...
+
+            $gitItems = Join-Path -Path $clonePath -ChildPath '.git*'
+            $ymlItems = Join-Path -Path $clonePath -ChildPath '*.yml'
+            Get-ChildItem -Path $gitItems, $ymlItems | Remove-Item -Recurse -Force
+        }
+        finally
+        {
+            pop-location
+        }
+}
```
Comment (daxian-dbw) [inline build.psm1]: nit: fix the indentation

**Review by daxian-dbw — APPROVED** [review]
LGTM. Left some comments

**Review by JamesWTruher — APPROVED** [review]
LGTM

File: .travis.yml (hunk 1 above)
Comment (TravisEz13) [reply .travis.yml]: Not sure what it was for.

Comment (TravisEz13) [reply .travis.yml]: I removed it and it appears to not be needed.  I added a comment to why my exclusion is there.

**Review by adityapatwardhan — APPROVED** [review]

**Comment by iSazonov** [comment]
Should we change our contribution docs too?

**Comment by TravisEz13** [comment]
Please feel free to file an issue

**Pull request merged**